pixl_dcmd/src/pixl_dcmd/main.py

```diff
@@ -81,344 +81,139 @@ def anonymise_dicom(dataset: Dataset) -> Dataset:
-                )
-                dataset[grp, el].value = new_age
-
-        # Change value into hash from hasher API.
-        elif op == "secure-hash":
-            if [grp, el] in dataset:
-                if grp == 0x0010 and el == 0x0020:  # Patient ID
-                    pat_value = mrn + accession_number
-
-                    hashed_value = _hash_values(grp, el, pat_value, hasher_host_url)
-                    # Query PIXL database
-                    existing_image = query_db(mrn, accession_number)
-                    # Insert the hashed_value into the PIXL database
-                    add_hashed_identifier_and_save(existing_image, hashed_value)
-                else:
-                    pat_value = str(dataset[grp, el].value)
-
-                    hashed_value = _hash_values(grp, el, pat_value, hasher_host_url)
-                if dataset[grp, el].VR == "SH":
-                    hashed_value = hashed_value[:16]
-
-                dataset[grp, el].value = hashed_value
-
-                message = f"Changing: {name} (0x{grp:04x},0x{el:04x})"
-                logger.debug(f"\t{message}")
-            else:
-                message = f"Missing: {name} (0x{grp:04x},0x{el:04x})\
-                 - Operation ({op})"
-                logger.debug(f"\t{message}")
-
-    return dataset
-
-
-def hash_endpoint_path_for_tag(group: bytes, element: bytes) -> str:
-    """Call a hasher endpoint depending on the dicom tag group and emement"""
-    if group == 0x0010 and element == 0x0020:  # Patient ID
-        return "/hash-mrn"
-    if group == 0x0008 and element == 0x0050:  # Accession Number
-        return "/hash-accession-number"
-
-    return "/hash"
-
-
-def _hash_values(grp: bytes, el: bytes, pat_value: str, hasher_host_url: str) -> str:
-    ep_path = hash_endpoint_path_for_tag(group=grp, element=el)
-    payload = ep_path + "?message=" + pat_value
-    request_url = hasher_host_url + payload
-    response = requests.get(request_url)
-    # All three hashing endpoints return application/text so should be fine to
-    # use response.text here
+    hasher_req_url = (
+        f"http://{HASHER_API_AZ_NAME}:{HASHER_API_PORT}/hash?message={pat_value}"
+    )
+    if hash_len:
+        hasher_req_url += f"&length={hash_len}"
+    response = requests.get(hasher_req_url)
     logger.debug("RESPONSE = %s}" % response.text)
     return response.text
+
+
+def enforce_whitelist(
+    dataset: Dataset, tag_scheme: list[dict], recursive: bool
+) -> None:
+    """
+    Enforce the whitelist on the dataset.
+    """
+    dataset.walk(lambda ds, de: _whitelist_tag(ds, de, tag_scheme), recursive)
+
+
+def _whitelist_tag(dataset: Dataset, de: DataElement, tag_scheme: list[dict]) -> None:
+    """Delete element if it is not in the tagging schemе."""
+    tag_dict = _scheme_list_to_dict(tag_scheme)
+    if (
+        # de.VR != "SQ"
+        # and (de.tag.group, de.tag.element) in tag_dict
```

Comment:
should probably just remove this, interesting that ruff didn't complain about it